src/main/java/de/unistuttgart/overworldbackend/service/CourseService.java

```diff
@@ -143,4 +148,87 @@ private Dungeon configureDungeon(int dungeonId, DungeonConfig dungeonConfig) {
+    Course course = courseRepository
+      .findById(id)
+      .orElseThrow(() ->
+        new ResponseStatusException(HttpStatus.NOT_FOUND, String.format("There is no course with id %s.", id))
+      );
+    List<World> worlds = new ArrayList<>();
+    course.getWorlds().forEach(world -> cloneWorld(world, worlds));
+
+    Course cloneCourse = new Course(
+      courseInitialData.getCourseName(),
+      courseInitialData.getSemester(),
+      courseInitialData.getDescription(),
+      false,
+      worlds
+    );
+    courseRepository.save(cloneCourse);
+    return courseMapper.courseToCourseDTO(cloneCourse);
+  }
+
+  private void cloneWorld(World oldWorld, List<World> newWorlds) {
+    Set<MinigameTask> minigameTasks = new HashSet<>();
+    Set<NPC> npcs = new HashSet<>();
+    List<Dungeon> dungeons = new ArrayList<>();
+    oldWorld.getMinigameTasks().forEach(minigameTask -> minigameTasks.add(cloneMinigameTask(minigameTask)));
+    oldWorld.getNpcs().forEach(npc -> npcs.add(cloneNPC(npc)));
+
+    oldWorld.getDungeons().forEach(dungeon -> cloneDungeon(dungeon, dungeons));
+
+    newWorlds.add(
+      new World(
+        oldWorld.getStaticName(),
+        oldWorld.getTopicName(),
+        false,
+        minigameTasks,
+        npcs,
+        dungeons,
+        oldWorld.getIndex()
+      )
+    );
+  }
+
+  private void cloneDungeon(Dungeon oldDungeon, List<Dungeon> newDungeons) {
+    Set<MinigameTask> minigameTasks = new HashSet<>();
+    Set<NPC> npcs = new HashSet<>();
+    oldDungeon.getMinigameTasks().forEach(minigameTask -> minigameTasks.add(cloneMinigameTask(minigameTask)));
+    oldDungeon.getNpcs().forEach(npc -> npcs.add(cloneNPC(npc)));
+    newDungeons.add(
+      new Dungeon(
+        oldDungeon.getStaticName(),
+        oldDungeon.getTopicName(),
+        false,
+        minigameTasks,
+        npcs,
+        oldDungeon.getIndex()
+      )
+    );
+  }
+
+  private NPC cloneNPC(NPC npc) {
+    List<String> text = new ArrayList<>();
+    npc.getText().forEach(npcText -> text.add(npcText));
+    return new NPC(text, npc.getIndex());
+  }
+
+  private MinigameTask cloneMinigameTask(MinigameTask minigameTask) {
+    if (minigameTask.getGame() == null) {
+      return new MinigameTask(null, null, minigameTask.getIndex());
+    }
+    if (minigameTask.getGame().equals("NONE")) {
+      return new MinigameTask("NONE", null, minigameTask.getIndex());
+    }
+    if (minigameTask.getGame().equals("CHICKENSHOCK")) {
+      ChickenshockConfiguration config = chickenshockClient.getConfiguration(minigameTask.getConfigurationId());
+      config.setId(null);
+      config.getQuestions().forEach(chickenshockQuestion -> chickenshockQuestion.setId(null));
+      config = chickenshockClient.postConfiguration(config);
+      MinigameTask newMinigame = new MinigameTask("CHICKENSHOCK", config.getId(), minigameTask.getIndex());
+      return newMinigame;
+    }
+    return minigameTask;
```

Comment:
I don't think it's a good idea to return the same instance when no counterpart could be found.
That'll most likely lead to **really** weird bugs when Hibernate then reuses an existing configuration, and you delete the original course. I think this configuration would then be broken forever.

Reply:
It would probably be a better idea to notify the user somehow, i.e. by sending a response which minigame configurations could not be cloned.